src/mainwindow.cpp

```diff
@@ -1156,37 +1167,37 @@ void MainWindow::changeEvent(QEvent* event) {
       ui.statusBar->hide();
       if(thumbnailsDock_)
         thumbnailsDock_->hide();
+    
       // NOTE: in fullscreen mode, all shortcut keys in the menu are disabled since the menu
       // is disabled. We needs to add the actions to the main window manually to enable the
       // shortcuts again.
       ui.menubar->hide();
-      const auto actions = ui.menubar->actions();
-      for(QAction* action : qAsConst(actions)) {
-        if(!action->shortcut().isEmpty())
-          addAction(action);
-      }
-      addActions(ui.menubar->actions());
+      toggleActions(false); 
+    
       ui.view->hideCursor(true);
+      ui.actionCompact->setDisabled(true);
     }
     else { // restore to normal window mode
       ui.view->setFrameStyle(QFrame::StyledPanel|QFrame::Sunken);
       ui.view->setBackgroundBrush(QBrush(app->settings().bgColor()));
       ui.view->updateOutline();
+    
       // now we're going to re-enable the menu, so remove the actions previously added.
-      const auto actions_ = ui.menubar->actions();
-      for(QAction* action : qAsConst(actions_)) {
-        if(!action->shortcut().isEmpty())
-          removeAction(action);
-      }
-      ui.menubar->show();
-      ui.toolBar->show();
-      if(ui.actionAnnotations->isChecked()){
+      toggleActions(false); 
+      
+      if(!ui.actionCompact->isChecked()){
+        ui.menubar->show();
+        ui.toolBar->show();
+        if(ui.actionAnnotations->isChecked()){
           ui.annotationsToolBar->show();
+        }
+        ui.statusBar->show();
+        if(thumbnailsDock_){
+          thumbnailsDock_->show();
```

Comment:
Thumbnail dock and annotation toolbar should be *outside* the block, as mentioned at https://github.com/lxqt/lximage-qt/pull/284#issuecomment-547873735

Also, there's no need to the method `toggleActions()`. Its name damages code readability too.

Also, the existence of shortcuts shouldn't be checked on adding/removing actions (→  https://github.com/lxqt/lximage-qt/pull/284#issuecomment-547873735)

Reply:
Ok @tsujan. I will review this again. I created the function to avoid rewriting the same code for the case of fullscreen and compact. I think I'll more explanation on what we want to achieve. For me is working good now. Can you explain me again please?

Reply:
> For me is working good now.

That's the problem: "For me". A code should work under all circumstances, not only when the app is used in a specific way.

> Can you explain me again please?

Please read my above explanations again. I've already explained some things twice.